packages/mobile/src/firebase/firebase.ts

```diff
@@ -218,39 +219,48 @@ export function appRemoteFeatureFlagChannel() {
     return null
   }
 
-  const errorCallback = (error: Error) => {
-    Logger.warn(TAG, error.toString())
-  }
-
   return eventChannel((emit: any) => {
-    const emitter = (snapshot: FirebaseDatabaseTypes.DataSnapshot) => {
-      const flags = snapshot.val()
-      Logger.debug(`Updated feature flags: ${JSON.stringify(flags)}`)
-      emit({
-        hideVerification: flags?.hideVerification ?? false,
-        showRaiseDailyLimitTarget: flags?.showRaiseDailyLimitTargetV2 ?? undefined,
-        celoEducationUri: flags?.celoEducationUri ?? null,
-        celoEuroEnabled: flags?.celoEuroEnabled ?? false,
-        shortVerificationCodesEnabled: flags?.shortVerificationCodesEnabled ?? false,
-        inviteRewardsEnabled: flags?.inviteRewardsEnabled ?? false,
-        inviteRewardCusd: flags?.inviteRewardCusd ?? 1,
-        inviteRewardWeeklyLimit: flags?.inviteRewardWeeklyLimit ?? 5,
-        walletConnectEnabled: flags?.walletConnectEnabled ?? false,
-        rewardsABTestThreshold:
-          flags?.rewardsABTestThreshold ?? '0xffffffffffffffffffffffffffffffffffffffff',
-        rewardsPercent: flags?.rewardsPercent ?? 5,
-        rewardsStartDate: flags?.rewardsStartDate ?? 1622505600000,
-        rewardsMax: flags?.rewardsMax ?? 1000,
-      })
+    const emitter = (fetchedRemotely: boolean) => {
+      if (fetchedRemotely) {
+        Logger.debug('Configs were retrieved from the backend and activated.')
+        const flags: FirebaseRemoteConfigTypes.ConfigValues = remoteConfig().getAll()
+        emit({
+          hideVerification: flags.hideVerification.asBoolean(),
+          showRaiseDailyLimitTarget: flags.showRaiseDailyLimitTargetV2.asString(),
+          celoEducationUri: flags.celoEducationUri.asString(),
+          celoEuroEnabled: flags.celoEuroEnabled.asBoolean(),
+          shortVerificationCodesEnabled: flags.shortVerificationCodesEnabled.asBoolean(),
+          inviteRewardsEnabled: flags.inviteRewardsEnabled.asBoolean(),
+          // TODO: why was this defaulting to 1 when the initial state is 0?
+          inviteRewardCusd: flags.inviteRewardCusd.asNumber(),
```

Comment:
good question! looks like firebase has a set defaults method so I will use that here to be safe. The weird thing is now we have 2 places on the client where we set defaults ...

Reply:
ah maybe I can use the initial state object to set defaults here so we have only one object

Reply:
do we need a migration for removing feature flags from the local state? maybe I will do that as a separate PR if so.

Reply:
ah I remember now about the defaults. I used the fetchedRemotely parameter which tells us whether the flags arrived from the backend. Otherwise no need to emit because we have defaults defined locally. That way we don't duplicate defaults here.

Reply:
> do we need a migration for removing feature flags from the local state? maybe I will do that as a separate PR if so.

I see the flags have the same name, so no migration needed.

Reply:
Sounds good!